src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -414,6 +457,21 @@ public Response get(String json) throws Exception
 
 			for (QueryMetric query : queries)
 			{
+				Map<String, Collection<String>> tags = query.getTags().asMap();
+				Set<String> keys = tags.keySet();
+
+				span.setTag("metric_name", query.getName());
+
+				for (String key : keys) {
+					StringBuilder str = new StringBuilder();
+					Iterator<String> iter = tags.get(key).iterator();
+					while(iter.hasNext()) {
+						str.append(iter.next() + ",");
+					}
+					str.deleteCharAt(str.lastIndexOf(","));
+					span.setTag(key, str.toString());
+				}
```

Comment:
I think I have added this comment before and may be it is lost. In case of Multiple queries we will overwrite tags in one span. This should move to the underlying layer with a separate span per query.

Reply:
Please double check it. This has been addressed. Tags are not replaced now. Have a look at this trace: https://app.lightstep.com/Production/trace?span_guid=3fcf7da57121b223&trace_id=606a0d4d2ade52f5&tracer_id=1e578c436264c056&operation_name=datapoints_query&at_micros=1530617531263003#span-3fcf7da57121b223

Just search by tag as key="," this will show keys with multiple tags in the drop down.

Reply:
Makes it un-searchable because it transforms the tag value to be very un deterministic + it doesn't show the query behavior.

Reply:
Again, once we have better search capability do we change the instrumentation again? For me the search works as we filter/autocomplete through the values. It's not the best but better than in logs or nothing. At least this provides some useful information in troubleshooting Kairosdb queries.

Reply:
>Actually this argument is in favor of instrumentation assuming least capabilities. We need to work on basis of how can we get the best value of our instrumentation assuming lowest features on the backend side.
It also helps better tagging/logging practice, since having those advanced regex based search for instance, could lead to jamming all values in one tag and ending up with a worse tracing + getting locked into single backend.

Reply:
@mohabusama Is there a better way that you can suggest here to include the keys with span tags? Or you want to include this in the span.logs.